### ISSUE TITLE
Bug in MigrateToURI3

### ISSUE THREAD

**Issue by danielkucera** [libvirt/libvirt-go#9]
Hi,
I'm setting following struct as param:
```
    migrParam := libvirt.DomainMigrateParameters{
        URISet: true,
        URI: dest_uri,
    }
err = domain_obj.MigrateToURI3(dest_uri, &migrParam, mFlags)
```
but when i look into libvirt debug, I see following params:
```
2017-03-07 16:54:09.356+0000: 27803: debug : virDomainMigrateUnmanagedParams:3432 : params["compression"]=(string)<null>
2017-03-07 16:54:09.356+0000: 27803: debug : virDomainMigrateUnmanagedParams:3432 : params["compression.mt.level"]=(int)0
2017-03-07 16:54:09.356+0000: 27803: debug : virDomainMigrateUnmanagedParams:3432 : params["destination_name"]=(string)<null>
2017-03-07 16:54:09.356+0000: 27803: debug : virDomainMigrateUnmanagedParams:3432 : params["graphics_uri"]=(string)<null>
2017-03-07 16:54:09.356+0000: 27803: debug : virDomainMigrateUnmanagedParams:3432 : params["compression.mt.threads"]=(int)0
2017-03-07 16:54:09.356+0000: 27803: debug : virDomainMigrateUnmanagedParams:3432 : params["compression.xbzrle.cache"]=(ullong)0
2017-03-07 16:54:09.356+0000: 27803: debug : virDomainMigrateUnmanagedParams:3432 : params["bandwidth"]=(ullong)0
2017-03-07 16:54:09.356+0000: 27803: debug : virDomainMigrateUnmanagedParams:3432 : params["listen_address"]=(string)<null>
2017-03-07 16:54:09.356+0000: 27803: debug : virDomainMigrateUnmanagedParams:3432 : params["disks_port"]=(int)0
2017-03-07 16:54:09.356+0000: 27803: debug : virDomainMigrateUnmanagedParams:3432 : params["compression.mt.dthreads"]=(int)0
2017-03-07 16:54:09.356+0000: 27803: debug : virDomainMigrateUnmanagedParams:3432 : params["destination_xml"]=(string)<null>
2017-03-07 16:54:09.356+0000: 27803: debug : virDomainMigrateUnmanagedParams:3432 : params["persistent_xml"]=(string)<null>
2017-03-07 16:54:09.356+0000: 27803: debug : virDomainMigrateUnmanagedParams:3432 : params["auto_converge.increment"]=(int)0
2017-03-07 16:54:09.356+0000: 27803: debug : virDomainMigrateUnmanagedParams:3432 : params["migrate_uri"]=(string)qemu+ssh://10.130.16.101/system
2017-03-07 16:54:09.356+0000: 27803: debug : virDomainMigrateUnmanagedParams:3432 : params["auto_converge.initial"]=(int)0
```

**Comment by danielkucera** [comment]
I suspect function typedParamsPackNew. I believe you should be checking if value is set:

```
func typedParamsPackNew(infomap map[string]typedParamsFieldInfo) (*[]C.virTypedParameter, error) {
        nparams := 0
        for _, value := range infomap {
            **if *value.set {**
                if value.sl != nil {
                        nparams += len(*value.sl)
                } else {
                        nparams++
                }
            }
        }

        cparams := make([]C.virTypedParameter, nparams)
        nparams = 0
        for key, value := range infomap {
            **if *value.set {**
                cfield := C.CString(key)
                defer C.free(unsafe.Pointer(cfield))
                clen := len(key) + 1
                if clen > C.VIR_TYPED_PARAM_FIELD_LENGTH {
                        clen = C.VIR_TYPED_PARAM_FIELD_LENGTH
                }

```

**Comment by berrange** [comment]
Yes, I think you are right. I'll figure out a test case to validate this

**Comment by berrange** [comment]
Pushed this fix to git master

commit 8147ec242032314bf09a81da07e72e64295450b7
Author: Daniel P. Berrange <berrange@redhat.com>
Date:   Wed Mar 8 11:51:50 2017 +0000

    Fix encoding of typed parameters which are not set
    
    Fixes bug #9
    
    Signed-off-by: Daniel P. Berrange <berrange@redhat.com>

it'll show up in this github mirror within the hour.
